### DISCUSSION TITLE
pyproject - Pinning versions

### DISCUSSION THREAD

**Discussion by lmilbaum** [fedora-infra/anitya#1617] [Ideas]
Updating poetry.lock file breaks the CI. The update is capturing too many updates which prevents a granular control on which package version lands in main.
Pinning the packages versions can provide the desired granularity.
Renovate can take care of updating the packages and the lockfile.

**Comment by Zlopez** [comment]
The renovate already updates the poetry.lock file, I just update it from time to time. I'm pinning the package versions just in cases, where it breaks Anitya, otherwise I'm trying to stay at the latest versions.

I'm actually not sure what are you trying to propose here?

**Reply by lmilbaum** [reply]
I think the following renovate key/value is required for updating the lock file:
"lockFileMaintenance" = true

For example: alembic is not up to date.

Would suggest to pin exact versions like this:

s/alembic = "^1.8.1"/alembic = "1.11.1"/

**Comment by Zlopez** [comment]
I don't want to pin to exact version if there is no good reason to it. In production deployment we use the Fedora packaged libraries where it's possible to get them and pinning to exact version could result that they are not found in Fedora.

Same for the dev environments. I don't want to get everything from PyPi just because the exact version is not yet packaged.

**Reply by lmilbaum** [reply]
Ah. I wasn't aware that the prod environment is different from the dev one.  Thanks for the clarification.
